alns/accept/LateAcceptanceHillClimbing.py

```diff
@@ -0,0 +1,86 @@
+from collections import deque
+
+from alns.accept.AcceptanceCriterion import AcceptanceCriterion
+
+
+class LateAcceptanceHillClimbing(AcceptanceCriterion):
+    """
+    The Late Acceptance Hill Climbing (LAHC) criterion accepts a candidate
+    solution when it is better than the current solution from a number of
+    iterations before.
+
+    This implementation is based on the description of LAHC in [1].
+
+    Parameters
+    ----------
+    n_iterations: int
+        Non-negative integer specifying the number of iterations of previous
+        current solution.
+        last objectives to store.
+        Default: 0 (no objectives stored).
+    on_improve: bool
+        Bool indicating whether or not to accept candidate solutions if they are
+        better than the current solution. Default: False.
+    only_better: bool
+        Bool indicating whether or not to only store current solutions that are
+        better than the previous current. Default: False
+
+    References
+    ----------
+    [1]: Burke, E. K., & Bykov, Y. The late acceptance hill-climbing heuristic.
+         *European Journal of Operational Research* (2017), 258(1), 70-78.
+    """
+
+    def __init__(
+        self,
+        n_iterations: int = 0,
+        on_improve: bool = False,
+        only_better: bool = False,
```

Comment:
Comment: I find the intuition behind these parameters a little confusing. I suspect that'll improve after refactoring. I might comment with a naming suggestion then.

Reply:
See newest version for different parameter names.